### DISCUSSION TITLE
OBD Kangoo ZE 2011 : reset DTC33d496 error

### DISCUSSION THREAD

**Discussion by Filtracafe** [cedricp/ddt4all#1117] [Q&A]
Hello, I am new to github. I live in France and I own a Kangoo ZE. I am looking for a developer who would be able to detect and clear the DTC33d496 error on my kangoo ZE 2011. I noticed that the canze application supports the error code on the zoe but that, on the other hand, the application does not is not developed for the kangoo. I am ready to finance the development project.

**Comment by AndyIreland** [comment]
hello there I could look remotely for you if you have:
A Bluetooth or Wired USB ELM327 diagnostic OBD2 adaptor (wifi obd2 wont work) 
A Windows 7/8/10/11 Laptop which can receive internet through either home router or hot-spot to phone

If the error is cleared though and the fault has not been rectified or part changed if needed the DTC code will either not be able to be cleared or will be cleared and possibly come back again .

**Comment by Filtracafe** [comment]
Thank you for your answer, I actually understand. 

Is this type of equipment "Konnwei Kw902" insufficient?

**Reply by AndyIreland** [reply]
if its Bluetooth and not wifi version I think it will be OK

**Reply by Janko1975** [reply]
Hi, this error code also appeared on an electric Renault Fluance. Can you tell me what the solution was in Kangoo, to see if it helps me too?

**Comment by Filtracafe** [comment]
Je n'ai malheureusement aucune solution à proposer pour ce code erreur...!!

**Reply by Janko1975** [reply]
Thank You.

**Reply by Falah-EVSolar** [reply]
Hi. I have the same DTC 33d496  on Renault Fluence ZE. Can you tell if you found what the solution was?

**Comment by Filtracafe** [comment]
Hi

Unfortunely i hvnt found any solution..!

Great day

Le sam. 1 mars 2025 à 04:27, Falah-EVSolar ***@***.***> a
écrit :

**Comment by matzel82** [comment]
> Hello, I am new to github. I live in France and I own a Kangoo ZE. I am looking for a developer who would be able to detect and clear the DTC33d496 error on my kangoo ZE 2011. I noticed that the canze application supports the error code on the zoe but that, on the other hand, the application does not is not developed for the kangoo. I am ready to finance the development project.


You probably need a new battery

After the guided diagnosis, I work at Renault, the code means that a new battery must be installed.

**Reply by Falah-EVSolar** [reply]
what to do if DTC33d496 error keeps showing after installing new traction battery for Renault Fluence ZE?

**Reply by TheSmartGerman** [reply]
As far as I got it, you have to replace cells and reset SOH value on BMS (LBC) EEPROM. You can buy a manual how to do this: https://evclinic.eu/evc-knowhow-academy/renault/bms/kangoo-ze-33d496-reset-operation-manual-190520251235123/

**Comment by Falah-EVSolar** [comment]
I have Renault Fluence ZE 2012.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by solar-pwr** [comment]
Renault's 33D496 = Nissan's P33D4. BMS DTC's are the same, just ignore the last 2 digits.

According to Nissan's manual, it's "BATTERY DETERIORATION DIAGNOSIS - BATTERY INTERNAL RESISTANCE DIAG". It shows up when SOH and Hx get too low. Can be caused by bad cells, but not necessarily. Disassembling the battery is inevitable to find the exact cause.

There are multiple ways to reset SOH/Hx on this battery. One of them is overwriting the checksums in the EEPROM of BMS #1. For Kangoo 22kWh battery, these are located at address 4-5 and 8-9 (16 bit values with least significant digit being always "A"). This causes both copies of battery data to become invalid, forcing the BMS to reload factory defaults from flash.

**Reply by TheSmartGerman** [reply]
Thx, for the explanation. There is some reverse engineering. Have a Look over here: https://www.renardspark.com/

To reset the EEPROM HX data, there is a direct way with **ddt** possible: 
- Open the LBC, look for all commands and have a look vor 2161 DataRead. 
-  Toggle to expert mode, 
-  now you can edit the **degradation values** back to 0.99 and change **Full capacity (AH)**. 
- **And write the data.** I guess it makes the same: Reload default settings from flash or calculates new: SOH will jump to 118%. Without new batteries it will drop fast again. But the DTC will be gone...
- after writing the lbc needs a power cycle (all ecus have to go to sleep)
- If writing not works immediately drive a few km without power cycle, do the power cycle afterwards.
- 
Note: there is SOH and Mileage inputs files, you can't modify. Or well, I will not overwritten. I guess to reset battery mileage you still have to access eeprom directly, set to zero and do the right checksum. Well, this will be interesting. After rebuilding a old battery pack, reset to zero and get the real mileage.  

Well with reset degradations values, the car will also use full power for acceleration and recuperation for a while. Until the SOH / HX value is back to the more or less real value. I did the reset, because my car was at 62% SOH, over the last 6 months it dropped back to 67%, so I gained +5% of SOH for the moment. Long-term goal, will be a battery swap. But with larger capacity, now I try to fine way how get the LBC handle more capacity than 44kWh...

**Reply by young-Gentleman** [reply]
Hi TheSmartGerman,
a friend of mine is struggling with his Kangoo ZE (22 kWh) with this error. I would kindly ask you for help, because we could not follow your instructions.
Today we set up DDT4ALL [v3.1.2] on a Windows 10 Laptop and managed to find a ecu.zip from 2023.
We used my Dongle Vgate iCar Pro 2S on the Android Bluetooth mode. We could connect to the car and also got connection to LBC.
When I tried everything on the desk, I also found the possbility to see every command that is defined in the json. There I found 21_61 and also the values for degradation. But when connected to the car, we only had "Dokumentation", "Bildschirm", "CAN-Sniffer".
Between your first four bullet points, we would need more detailed instructions how you proceeded. We could find "SOHE functionality" and below that there was listed "LID 70" where we could read out "70_#05_Degradation (K)" and "70_#07_Degradation (K-1)". But all fields were filled with "NO DATA %"
Probably we did use a different json than you did. We tried "X61_22kWh_EV_BMS" first, followed by "X10_X61_X62_L38_EV_BMS_v6.3_LR" and then the slightly different variants of that file: 6.3, 6.4, 6.4.1.
What is needed to be able to write into the ECU? We used Extended Diagnostic Mode but the return from the ECU was "not supported".
I think we are nearly at the right point, but we didn't find the last issue, I guess.
Any help is greatly appreciated!

**Reply by TheSmartGerman** [reply]
Well, in fact I use Pyren3 in DDT-Mode... I do have Version 4.4 for the LBC. May you need a older ECU Database... ddt4all or pyren3 use the same database. Even ECUTweaker for mobile phones.

<img width="1196" height="576" alt="grafik" src="https://github.com/user-attachments/assets/330438f8-bfb8-4a9d-8aa1-9b76c840840c" />

You should have ddt_all_command list

<img width="477" height="570" alt="grafik" src="https://github.com/user-attachments/assets/9531dd3c-690f-40e9-8c92-43f1c8cf892c" />

In Expert Mode your are able to write

<img width="1515" height="742" alt="grafik" src="https://github.com/user-attachments/assets/0ad58a93-9768-43df-83f8-19bf31d9fcc7" />



[R_L38_X61_EV_LBC_v4.4_20140407T155120.xml](https://github.com/user-attachments/files/30676849/R_L38_X61_EV_LBC_v4.4_20140407T155120.xml)

**Comment by young-Gentleman** [comment]
Hello,I installed Python 3 and PyRen, so far this runs.Since I only have a DDT4ALL database (ecu.zip with thousands of jsons), your xml of the LBC was very helpful. But we are missing a project file and eculist.xml.We do not want to do any other diagnosis than tweaking the SoH value for this single car. If you could send me those two missing files, this would be sufficient, I think.Best regards, ChristianAm 03.08.2026 um 21:47 schrieb TheSmartGerman ***@***.***>:﻿
Well, in fact I use Pyren3 in DDT-Mode... I do have Version 4.4 for the LBC. May you need a older ECU Database... ddt4all or pyren3 use the same database. Even ECUTweaker for mobile phones.
grafik.png (view on web)
You should have ddt_all_command list
grafik.png (view on web)
In Expert Mode your are able to write
grafik.png (view on web)
R_L38_X61_EV_LBC_v4.4_20140407T155120.xml

**Reply by TheSmartGerman** [reply]
Try this database: https://github.com/cedricp/ddt4all/discussions/1683#discussioncomment-15815169

I did some merging...  This is my ecu list, what do you mean with project file?

[eculist.xml](https://github.com/user-attachments/files/30716750/eculist.xml)

**Reply by young-Gentleman** [reply]
We finally got it to work.
We could not succeed with Pyren, but we found away in DDT4ALL to read the respective values and to re-write them:
<img width="1008" height="756" alt="IMG_1095_clean" src="https://github.com/user-attachments/assets/0d2269d6-ef97-4b75-8f7d-b8108875a54e" />
First we read out the values of Read_Data_by_Identifier (21_61) which gave us SoH, Mileage, Sum kWh, two values for degradation coeff., and two "done" flags for the coeff initial value set.
<img width="1008" height="756" alt="IMG_1094_clean" src="https://github.com/user-attachments/assets/9b086adc-331d-4842-8702-6d589bdcadae" />
Then we used the button for custom commands and chose "DataWrite.3B61". This was the exactly fitting mask for the values that we just read out. In the mask all needed Keys and Data types were provided and we filled in the just read out values (screenshotted them before).
Rahmen berechnen "calculate frame" converted everything into one long HEX frame and we sent it out: degradation to 0.99, mileage and sum of kWh to the original values (again), flags set to "done".
We sent it and received a positive response.
Reading again: the newly programmed values were persistent.
We cycled T15 off and on, the error was still present.
Then we locked the vehicle and waited for 10 minutes.
Unlocking doors, T15 on: error was gone.

Thank you [TheSmartGerman](https://github.com/TheSmartGerman) for your support, we couldn't have found it without you!

**Reply by TheSmartGerman** [reply]
You also can reset data read 2166 and 2167. It's the charge counter and drive/park history. Not important. But may interesting how fast SOH drops again.

May I also have to check DDT4all again... never found this screen.

**Reply by young-Gentleman** [reply]
We had only a "ecu.zip" which contains all jsons inside. We could not get it to work with Pyren. Your XML for X61_LBC would have been useful, but since we only had this one file, Pyren wasn't providing us a list of vehicles nor ECUs to choose from. I thought, I would need some very basic xml for one vehicle "X61" with one ECU "LBC" to get Pyren to show us something to select from.
But now it is already solved in a different way.

**Reply by TheSmartGerman** [reply]
great! Am Ende wird alles gut ;)

**Reply by young-Gentleman** [reply]
The vehicle got its "Hauptuntersuchung" and can drive on the roads for two more years.

**Reply by solar-pwr** [reply]
I'm coming little bit too late, you've already found the proper way to do it :)
So, for others who may need this in future, the command to reset SOH on Renault's 22kWh battery should look like this:

3b6127102710000a0280df00ffffff000060032710271000

Mode must be set to 10c0.

**Reply by TheSmartGerman** [reply]
![FB_IMG_1785785274189.jpg](https://github.com/user-attachments/assets/4a5c7e8e-4ea4-4b43-ad7a-dd85dee383d5)

 Can ddt4all run Marcos?

**Reply by KarelSvo** [reply]
> So, for others who may need this in future, the command to reset SOH on Renault's 22kWh battery should look like this:
> 
> 3b6127102710000a0280df00ffffff000060032710271000

Your byte string contains data that cannot be modified and does not match that of other vehicles. If you use this string with other vehicles, the ECU will respond negatively to the modification request with code 7F3B22 (NR: Conditions Not Correct Or Request Sequence Error).

**Reply by solar-pwr** [reply]
Of course it must be the exact same type of battery (Kangoo 22kWh). But there's nothing related to a particular vehicle in that string. These are just the factory values which the firmware uses when the EEPROM is corrupt. That 7F3B22 error is related to two 32-bit values in the string (00ffffff and 00006003). I don't remember exactly what these are, but it does not allow decreasing these values. They can only be the same or higher. Maybe mileage and total kWh? Ahh, that makes sense. Then just copy these two values from the 2161 response.

**Reply by KarelSvo** [reply]
> Then just copy these two values from the 2161 response.

An even simpler method is to use a preparatory script, as shown in the photo above. It automatically inserts the bytes to be modified, making it universally applicable. There is no need to search for parameters within the ECU screens, making it a safe option for the occasional programmer.

**Reply by 85jcz5cwnh-ui** [reply]
Hola pero esta es la parte software antes se debe cambiar la celda resistiva , yo estoy reparando en este momento batería Renault Kangoo ez 22kw

**Reply by TheSmartGerman** [reply]
> Hola pero esta es la parte software antes se debe cambiar la celda resistiva , yo estoy reparando en este momento batería Renault Kangoo ez 22kw

sure, but in some countries you have  periodic technical inspection. Some are quite strict. And if there is a HV error, you'll not pass there test. Because it's not "safe" to be driven on public roads. Sooo... the car is scrap. BUT if you reset, error is gone. No one cares if you only can drive 50km with your remaining battery power... Sure new cells would be perfect!

See this: https://www.facebook.com/evclinic/posts/552-kwh-kangoo-ze-2015-battery-pack-has-been-born-with-300km-rangethis-is-the-wa/1100273422343835/
